src/test/java/org/synyx/urlaubsverwaltung/publicholiday/PublicHolidayApiControllerTest.java

```diff
@@ -33,135 +42,170 @@
+    void getPublicHolidays() throws Exception {
 
         final Settings settings = new Settings();
         settings.getWorkingTimeSettings().setFederalState(BADEN_WUERTTEMBERG);
-        when(settingsServiceMock.getSettings()).thenReturn(settings);
+        when(settingsService.getSettings()).thenReturn(settings);
 
-        perform(get("/api/holidays")
-            .param("year", "2016"))
-            .andExpect(status().isOk());
+        final LocalDate from = LocalDate.of(2016, 5, 19);
+        final LocalDate to = LocalDate.of(2016, 5, 20);
+        final Holiday fromHoliday = new Holiday(from, "", OFFICIAL_HOLIDAY);
+        final Holiday toHoliday = new Holiday(to, "", OFFICIAL_HOLIDAY);
+        when(publicHolidayService.getHolidays(from, to, BADEN_WUERTTEMBERG)).thenReturn(List.of(fromHoliday, toHoliday));
+
+        when(publicHolidayService.getWorkingDurationOfDate(from, BADEN_WUERTTEMBERG)).thenReturn(ONE);
+        when(publicHolidayService.getAbsenceTypeOfDate(from, BADEN_WUERTTEMBERG)).thenReturn(DayLength.MORNING);
+        when(publicHolidayService.getWorkingDurationOfDate(to, BADEN_WUERTTEMBERG)).thenReturn(TEN);
+        when(publicHolidayService.getAbsenceTypeOfDate(to, BADEN_WUERTTEMBERG)).thenReturn(DayLength.NOON);
 
-        verify(publicHolidayServiceMock).getHolidays(2016, BADEN_WUERTTEMBERG);
+        perform(get("/api/holidays")
+            .param("from", "2016-05-19")
+            .param("to", "2016-05-20"))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType("application/json"))
+            .andExpect(jsonPath("$.publicHolidays").exists())
+            .andExpect(jsonPath("$.publicHolidays", hasSize(2)))
+            .andExpect(jsonPath("$.publicHolidays[0].date", is("2016-05-19")))
+            .andExpect(jsonPath("$.publicHolidays[0].dayLength", is(1)))
+            .andExpect(jsonPath("$.publicHolidays[0].absencePeriodName", is("MORNING")))
+            .andExpect(jsonPath("$.publicHolidays[1].date", is("2016-05-20")))
+            .andExpect(jsonPath("$.publicHolidays[1].dayLength", is(10)))
+            .andExpect(jsonPath("$.publicHolidays[1].absencePeriodName", is("NOON")));
     }
 
     @Test
-    void ensureReturnsCorrectPublicHolidaysForYearAndMonth() throws Exception {
-
-        final Settings settings = new Settings();
-        settings.getWorkingTimeSettings().setFederalState(BADEN_WUERTTEMBERG);
-        when(settingsServiceMock.getSettings()).thenReturn(settings);
-
+    void getPublicHolidaysForInvalidPeriod() throws Exception {
         perform(get("/api/holidays")
-            .param("year", "2016")
-            .param("month", "4"))
-            .andExpect(status().isOk());
-
-        verify(publicHolidayServiceMock).getHolidays(2016, 4, BADEN_WUERTTEMBERG);
+            .param("from", "2016-01-01")
+            .param("to", "2015-01-01"))
+            .andExpect(status().isBadRequest());
```

Comment:
brauchen wir noch einen `MockUser` mit der office rolle? damit es auch wirklich um die `period` geht in dem test?